nosetimer/plugin.py

```diff
@@ -128,13 +128,13 @@ def _parse_time(self, value):
         """
         try:
             # Default time unit is a second, we should convert it to milliseconds.
-            return int(value) * 1000
+            return float(value) * 1000
         except ValueError:
             # Try to parse if we are unlucky to cast value into int.
             m = self.time_format.match(value)
             if not m:
                 raise ValueError("Could not parse time represented by '{t}'".format(t=value))
-            time = int(m.group('time'))
+            time = float(m.group('time'))
```

Comment:
Does this change the comparison logic now that time is a float? (e.g. I would assume that a test that runs in 5.5s might be flaky if a threshold of 5.5s is passed b/c of floating point math)

Reply:
Maybe this can be helpful, but I wonder if we should allow arbitrary precision vs using `decimal`. https://stackoverflow.com/a/33024979/133514

Reply:
If test is borderline, I am ok to fail it. Again, I don't see a real issue

Reply:
@yarikoptic the problem here that it might be flaky. I think we need to use Decimal module to handle this case properly as @mahmoudimus mentioned.

Reply:
sorry for being slow, could you please digest it to me even more ;)?  What I meant is that:
- unlikely you would like to time up really precisely to some low millisecond -- tests timing would anyways vary depending on the specific box (so vary across boxes), time of the day (cron jobs busy at some point but not another) etc.  So there is inherent "flakiness" in timing the tests anyways.
- `5.5` will have the same precision in machine representation on the same box, so even if something floating point number gets computed slightly below the threshold and then "rounded" into the threshold value thus failing -- imho good to know (see above) and fail.  If something above threshold gets "rounded" into the threshold -- also good. 

I do not think introducing some "margin" for comparison with "close" based on e.g. relative difference would really give any piece of mind in this case of comparing timings (since they vary anyways as per above etc) which are just threshold values.  Or am I missing some specific use case?

Reply:
@yarikoptic maybe you are right, we do not have to be so precise. We can adjust that in the future if needed.